### PR TITLE
Fixes #1314 - CONTRIBUTING Guidelines cleanup

### LINKED ISSUE #1314: CONTRIBUTING Guidelines fresh eye review and modifications by @zoepage
@zoepage has started to review the Contributing guidelines.

Opening this issue so she can commit step by step her changes on the document.

Also found these previous issues which are worth reviewing too.

* [x] #766 Contributing guide: npm install
* [ ] #1134 CONTRIBUTION Guidelines: clarify the syntax of commits messages and pull requests 
* [x] #1176 contributing guidelines - npm run init

--- edit by @zoepage ---
* [x] `sudo npm run script` -> get rid of sudo
* [x] `npm install -g grunt` -> get rid of global install
* [x] add login step to authorize application at [configuring-the-server section](https://github.com/webcompat/webcompat.com/blob/master/CONTRIBUTING.md#configuring-the-server)
* [x]  fix https://github.com/settings/applications/new

### PR DESCRIPTION
**Pull request by zoepage**
See #1314 

Overall list:
* [x] #766 Contributing guide: npm install
* [x] #1134 CONTRIBUTION Guidelines: clarify the syntax of commits messages and pull requests 
* [x] #1176 contributing guidelines - npm run init
* [x] `sudo npm run script` -> get rid of sudo
* [x] `npm install -g grunt` -> get rid of global install
* [x] add login step to authorize application at [configuring-the-server section](https://github.com/webcompat/webcompat.com/blob/master/CONTRIBUTING.md#configuring-the-server)

### CODE REVIEW

**Comment by miketaylr** [comment]
Sad, our tests are being flaky again. :(

(unrelated to these changes)

**Comment by zoepage** [comment]
Yes. It's again this one that fails :/

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -337,19 +369,23 @@ You can now edit `secrets.py` and
...
 
-The [instructions for creating a personal access token](http://help.github.com/articles/creating-an-access-token-for-command-line-use) are given on GitHub. Select public_repo to grant access to the public repositories through the personal access token.  Once you have created the token you can add it in the variable `OAUTH_TOKEN = ""` (yes, even if you're using your own credentials we still refer to it as a bot). More advanced users might want to create an environment variable called `OAUTH_TOKEN`. Either way is fine.
+  The [instructions for creating a personal access token](http://help.github.com/articles/creating-an-access-token-for-command-line-use) are given on GitHub. Select public_repo to grant access to the public repositories through the personal access token.  Once you have created the token you can add it in the variable `OAUTH_TOKEN = ""` (yes, even if you're using your own credentials we still refer to it as a bot). More advanced users might want to create an environment variable called `OAUTH_TOKEN`. Either way is fine.
```
Comment (miketaylr) [inline CONTRIBUTING.md]: I think we can remove the line "(yes, even if you're using your own credentials we still refer to it as a bot)" -- this variable used to be called `BOT_OAUTH_TOKEN`. I guess we forgot to update that line when we changed it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -375,7 +411,7 @@ You should now have a local instance of the site running at `http://localhost:50
 
 After certain kinds of changes are made, you need to build the project before serving it from a webserver will work
 
-* CSS: a build will run cssnext, combine custom media queries, and concat all source files into webcompat.dev.css. You'll need to re-build the CSS to see any changes, so it's recommended to use a watch task (see `make watch` or `grunt watch`).
+* CSS: a build will run cssnext, combine custom media queries, and concat all source files into webcompat.dev.css. You'll need to re-build the CSS to see any changes, so it's recommended to use a watch task (see `make watch` or `npm run watch`).
```
Comment (miketaylr) [inline CONTRIBUTING.md]: We removed the Makefile in https://github.com/webcompat/webcompat.com/commit/adc48ef679c1009a987b58f2ab3dcbc75155171e, so we can remove the `make watch` command.

**Comment by zoepage** [comment]
@miketaylr done!

**Pull request merged**